src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -973,8 +990,14 @@ public static boolean addItemsToItemHandler(final IItemHandler handler,
                                                 final boolean simulate,
                                                 final List<ItemStack> items) {
         // determine if there is sufficient room to insert all items into the target inventory
-        final boolean canMerge = simulateItemStackMerge(items, handler);
-
+        boolean canMerge;
+        if (items.size() == 1) {
+            canMerge = ItemHandlerHelper.insertItemStacked(handler, items.get(0), true).isEmpty();
+        } else {
+            // merging 2 or more stacks may change the slots available for merging,
+            // needing full simulation of the state of the inventory after every merge
+            canMerge = InventoryUtils.simulateItemStackMerge(items, handler);
+        }
```

Comment:
Not quite sure about the purpose of this change. Merging 1 stack of items can change the available slots for merging, if the input stack merges into a partial stack, filling it up, and then creating a new partial stack that has to be placed in a new slot.

Reply:
if we only have 1 stack to merge, the simulated insertion will mirror the actual result of the real insertion.
with 2 or more, the insertion of stack 1 can change the space available for stack 2.

Reply:
you're right